### PR TITLE
Refactor for improved usability and readability

### LINKED ISSUE #17: Streamline preprocessing pipeline
Data preprocessing is currently split into multiple steps, i.e.,

1. Download the datasets (where?).
2. Run `run.preprocess_dataset`.
3. Write the corresponding `*.lab` files using `run.export_mfa`.
4. Download Montreal Forced Aligner (MFA) and the corresponding CMU phonetic dictionary.
5. Run MFA (`mfa_align`) over the speech corpus.
6. Convert the output TextGrids to our `jsonl` format (`run.attach_mfa_alignment`).

We should make this process easier and document it somewhere.

### PR DESCRIPTION
**Pull request by daemon**
- Streamline data preprocessing pipeline (close #17)
- Improve class structure and code organization
- Remove unnecessary dependency on DeepSpeech

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,35 +1,48 @@
...
-python -m ww4ff.run.export_honkling -i workspace/<workspace_name>/model-best.pt -o <output_filename>.js --name RES8
+7. Attach the MFA alignment to the positive dataset:
+```bash
+DATASET_PATH=data/fire-positive python -m howl.run.attach_alignment --align-type mfa -i output-folder
 ```
+
+### Training and Running a Model
+
+1. Source the relevant environment variables for training the `res8` model: `source envs/res8.env`.
+2. Train the model: `python -m howl.run.train -i data/fire-negative data/fire-positive --model res8 --vocab fire --workspace workspaces/fire-res8`.
```
Comment (ljj7975) [inline README.md:47]: do we no longer need pretraining?

File: README.md [hunk 2]
```diff
@@ -1,35 +1,48 @@
...
+### Installation
 
-* [Install docker](https://docs.docker.com/engine/install/) and [enable GPU support](https://cnvrg.io/how-to-setup-docker-and-nvidia-docker-2-0-on-ubuntu-18-04/)
+A proper Pip package is coming soon. 
 
-* `docker build -t ww4ff .`
+1. `git clone https://github.com/castorini/howl && cd howl`
 
-* `./start.sh` will start up the docker. (if docker container is running it will use the same container) Note that /data and current directory will be mounted
+2. Install [PyTorch](https://pytorch.org) by following your platform-specific instructions.
```
Comment (ljj7975) [inline README.md:12]: what about deep speech? after MFA we no longer require it?

deep speech had to be installed explicitly was quite complicated to install the right thing. if we need it. can you add instruction

File: envs/res8.env [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
 export LEARNING_RATE=0.001
-export LR_DECAY=0.92
+export LR_DECAY=0.955
 export BATCH_SIZE=16
 export MAX_WINDOW_SIZE_SECONDS=0.5
-export USE_NOISE_DATASET=True
+export USE_NOISE_DATASET=False
 export NUM_MELS=40
-export INFERENCE_SEQUENCE=[0,1,2]
+export INFERENCE_SEQUENCE=[0]
```
Comment (ljj7975) [inline envs/res8.env:9]: what has it become [0] now?

File: README.md (hunk 1 above)
Comment (daemon) [reply README.md:47]: I didn't include that in the directions for simplicity.

File: README.md (hunk 2 above)
Comment (daemon) [reply README.md:12]: DeepSpeech is no longer needed for alignment.

File: envs/res8.env (hunk 3 above)
Comment (daemon) [reply envs/res8.env:9]: For the quickstart guide. Internally, we'll set it to `[0,1,2]`.

**Pull request merged**